src/core/PasswordHealth.cpp

```diff
@@ -21,10 +21,33 @@
 #include "PasswordHealth.h"
 #include "zxcvbn.h"
 
+namespace
+{
+    const static int ZXCVBN_ESTIMATE_THRESHOLD = 256;
+} // namespace
+
 PasswordHealth::PasswordHealth(double entropy)
-    : m_score(entropy)
-    , m_entropy(entropy)
 {
+    init(entropy);
+}
+
+PasswordHealth::PasswordHealth(const QString& pwd)
+{
+    auto entropy = 0.0;
+    auto pwdBytes = pwd.toUtf8();
+    entropy += ZxcvbnMatch(pwdBytes.left(ZXCVBN_ESTIMATE_THRESHOLD), nullptr, nullptr);
```

Comment:
Will this work with multi-byte UTF-8 chars? Maybe it's better to split based on character count and not on bytes.

Reply:
Zxcvbn operates on bytes, so no we don't get a choice here.

Reply:
Technically you'd only have multibyte unicode characters if you pasted them into the password field. We don't support generating passwords with unicode. But yes, we can do the limit check on the string, then convert it to utf8 bytes for zxcvbn.